### PR TITLE
Feat: Support SNS pagination

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

We only fetch one page of the SNS aggregator.

When there are more than 10 SNSes, the latest never appear.

# Changes

* Get the next page if the SNS Aggregator returns 10 SNSes.

# Tests

* Test the new functionality of the sns aggregator api.

### CODE REVIEW

**Comment by lmuntaner** [comment]
@dskloetd @mstrasinskis please review

File: frontend/src/tests/lib/api/sns-aggregator.api.spec.ts [hunk 1]
```diff
@@ -7,27 +7,56 @@ describe("sns-aggregator api", () => {
...
+      expect(mockFetch).toHaveBeenCalledWith(
+        `https://5v72r-4aaaa-aaaaa-aabnq-cai.small12.testnet.dfinity.network/v1/sns/list/page/1/slow.json`
+      );
+      expect(mockFetch).toHaveBeenCalledTimes(2);
     });
 
     it("should convert response", async () => {
       const mockFetch = jest.fn();
-      mockFetch.mockReturnValueOnce(
+      const [first, ..._rest] = aggregatedSnses;
```
Comment (dskloetd) [inline frontend/src/tests/lib/api/sns-aggregator.api.spec.ts:55]: `const first = aggregatedSnses[0]` ?

File: frontend/src/tests/lib/api/sns-aggregator.api.spec.ts [hunk 2]
```diff
@@ -7,27 +7,56 @@ describe("sns-aggregator api", () => {
...
+      mockFetch.mockReturnValue(
         Promise.resolve({
           ok: true,
-          json: () => Promise.resolve(aggregatedSnses),
+          json: () => Promise.resolve([first]),
```
Comment (dskloetd) [inline frontend/src/tests/lib/api/sns-aggregator.api.spec.ts:59]: Why did you change this to only return the first?
Also, it's not clear from the test why that should be equal to `aggregatorSnsMock`.

**Review by dskloetd — APPROVED** [review]
Code looks good and tests look correct but too confusing.
Approving if you promise to follow up to improve the tests.

File: frontend/src/tests/lib/api/sns-aggregator.api.spec.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/api/sns-aggregator.api.spec.ts:55]: true, that was a weird way of doing it, haha

File: frontend/src/tests/lib/api/sns-aggregator.api.spec.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/api/sns-aggregator.api.spec.ts:59]: It is not clear, yes. I can improve that in another PR.

File: frontend/src/tests/lib/api/sns-aggregator.api.spec.ts [hunk 3]
```diff
@@ -7,27 +7,56 @@ describe("sns-aggregator api", () => {
     afterEach(() => {
       jest.resetAllMocks();
     });
-    it("should fetch json", async () => {
+    it("should fetch json once if less than 10 SNSes", async () => {
       const mockFetch = jest.fn();
+      const [_first, ...rest] = aggregatedSnses;
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/lib/api/sns-aggregator.api.spec.ts]: renamed to make it clearer

**Comment by lmuntaner** [comment]
> Code looks good and tests look correct but too confusing. Approving if you promise to follow up to improve the tests.

Pinky swear 👼
